### ISSUE TITLE
Sac model perpendicular plane with normals missing

### ISSUE THREAD

**Issue by 777Gerald777** [PointCloudLibrary/pcl#5211]
There is a SACMODEL_PERPENDICULAR_PLANE but no SACMODEL_NORMAL_PERPENDICULAR_PLANE, should be easy to add it to pcl/sample_consensus/model_types.h.

**Comment by 777Gerald777** [comment]
setModelConstraints sounds cool but the function pcl::SACSegmentationFromNormals::getModel returns an empty NULL pointer before calling the segment execution, so i have to make an own class which inherits from pcl::SACSegmentationFromNormals to override initSACModel function so that i can use setModelConstraints. And also this complicated method does not work because initSACModel is not virtual, lol. so i have to inherit from SACSegmentation and copy code from SACSegmentationFromNormals....

ya, those git stuff with pull and push requests is too complicated for me, sorry (i only work with svn)... only wanted to make a feedback that there was something forgotten.....

**Comment by mvieth** [comment]
Yes, using `setModelConstraints` with `SACSegmentation` or `SACSegmentationFromNormals` is currently not easy. But it is when using [this other way for sample consensus](https://pcl.readthedocs.io/projects/tutorials/en/master/random_sample_consensus.html).

**Comment by 777Gerald777** [comment]
i found also something funny:

SACMODEL_PERPENDICULAR_PLANE behaves like SACMODEL_NORMAL_PARALLEL_PLANE.....

so, either, SACMODEL_NORMAL_PARALLEL_PLANE should be called SACMODEL_NORMAL_PERPENDICULAR_PLANE or
SACMODEL_PERPENDICULAR_PLANE  and SACMODEL_PARALLEL_PLANE should be switched, otherwise the term PERPENDICULAR and PARALLEL does not mean the same on the normal variant, lol

**Comment by mvieth** [comment]
Yes, but that is well documented, see for example [here](https://pointclouds.org/documentation/classpcl_1_1_sample_consensus_model_normal_parallel_plane.html):
`... In addition, the plane normal must lie parallel to a user-specified axis. This means that the plane itself will lie perpendicular to that axis, similar to SACMODEL_PERPENDICULAR_PLANE  ...`
I agree that the naming can be confusing, but we can't change that now without breaking the code/programs from users.

**Comment by 777Gerald777** [comment]
i know that it will break code, but i currently implemented my own "SAC_MODEL_PERPENDICULAR_PLANE" and debugged some hours of finding out why my plane was the wrong orientation....

i think that when i want to "upgrade" my algorithm from SAC_MODEL_PARALLEL to SAC_MODEL_NORMAL_PARALLEL i should expect that the plane orientation of the result does not flip... 

i dont know what a beautiful solution would be, maybe adding SAC_MODEL_NORMAL_PARALLEL2 and making/marking SAC_MODEL_NORMAL_PARALLEL as depricated and remove in future releses and make the naming consistent....

**Comment by mvieth** [comment]
> i dont know what a beautiful solution would be, maybe adding SAC_MODEL_NORMAL_PARALLEL2 and making/marking SAC_MODEL_NORMAL_PARALLEL as depricated and remove in future releses and make the naming consistent....

That would be a thing we could consider, but I don't see a nice way to deprecate the enum value (the [deprecated attribute](https://en.cppreference.com/w/cpp/language/attributes/deprecated) appears to only work for that since C++17, PCL aims to also compile with C++14), and there are more things to rename/deprecate than just the enum values, e.g. sac_model_normal_parallel_plane.h[pp] and the classes inside. I fear that trying to rename everything might just cause more confusion than the current state.

**Comment by 777Gerald777** [comment]
dont worry, for me it is ok. i only wanted to feedback it and provide possible solutions...
